lib/ancestry/class_methods.rb

```diff
@@ -155,26 +155,26 @@ def restore_ancestry_integrity!
           # For each node ...
           scope.find_each do |node|
             # ... rebuild ancestry from parents array
-            ancestry, parent = nil, parents[node.id]
+            ancestor_ids, parent = [], parents[node.id]
             until parent.nil?
-              ancestry, parent = if ancestry.nil? then parent else "#{parent}/#{ancestry}" end, parents[parent]
+              ancestor_ids, parent = [parent] + ancestor_ids, parents[parent]
```

Comment:
I tried making that change, and got an error:
```
NoMethodError: undefined method `id' for 1:Integer
```

Which is good for 2 reasons.

1. that means this code is actually tested
2. that means that parent contains an integer (the id) and not an object.

Should I assume a followup reponse that `parent` should be renamed to `parent_id`?

Reply:
> Should I assume a followup response that `parent` should be renamed to `parent_id`?

I guess if you are pushing for a new version change, I guess that might not be a bad idea since that is the generally the convention, but I wouldn't say it is a requirement now knowing that this method breaks the convention.